File: services/identificacao-api/3-Domain/Identificacao.Domain/Entities/Rubrica.cs
namespace Identificacao.Domain.Entities;

public class Rubrica
{
    public Guid Id { get; private set; }
    public string Sigla { get; private set; } = string.Empty;
    public string Nome { get; private set; } = string.Empty;
    public bool ExigeClassificacao { get; private set; }

    private Rubrica() { } // Para o EF Core

    public static Rubrica Criar(Guid id, string sigla, string nome, bool exigeClassificacao)
    {
        return new Rubrica
        {
            Id = id,
            Sigla = sigla,
            Nome = nome,
            ExigeClassificacao = exigeClassificacao
        };
    }
}

File: services/arrecadacao-api/arrecadacao-domain/src/main/java/br/com/ecad/arrecadacao/domain/entities/Rubrica.java
package br.com.ecad.arrecadacao.domain.entities;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import java.util.Objects;
import java.util.UUID;

@Entity
@Table(
        name = "rubricas",
        schema = "arrecadacao",
        uniqueConstraints = @UniqueConstraint(name = "uq_rubricas_sigla", columnNames = "sigla"))
public class Rubrica {
    @Id
    private UUID id;

    @Column(nullable = false, length = 20)
    private String sigla;

    @Column(nullable = false, length = 100)
    private String nome;

    @Column(name = "exige_classificacao", nullable = false)
    private boolean exigeClassificacao;

    protected Rubrica() {
    }

    public Rubrica(UUID id, String sigla, String nome, boolean exigeClassificacao) {
        this.id = Objects.requireNonNull(id, "id must not be null");
        this.sigla = Objects.requireNonNull(sigla, "sigla must not be null");
        this.nome = Objects.requireNonNull(nome, "nome must not be null");
        this.exigeClassificacao = exigeClassificacao;
    }

    public UUID getId() {
        return id;
    }

    public String getSigla() {
        return sigla;
    }

    public String getNome() {
        return nome;
    }

    public boolean isExigeClassificacao() {
        return exigeClassificacao;
    }
}
